Source/Engine/UI/GUI/Common/TextBox.cs

```diff
@@ -155,7 +155,11 @@ public override void DrawSelf()
             if (IsMouseOver || IsNavFocused)
                 backColor = BackgroundSelectedColor;
             Render2D.FillRectangle(rect, backColor);
-            Render2D.DrawRectangle(rect, IsFocused ? BorderSelectedColor : BorderColor);
+            
+            var borderColor = Border.Color != Color.Transparent ? Border.Color : BorderColor;
```

Comment:
This feels sus. When I set Border.Color to transparent, I'd kinda expect it to end up being transparent.

Reply:
I was trying to find a way to make the border changes backwards compatible with the way it is currently done, although I realized it needs some changes to work correctly